plugins/org.eclipse.emf.ecore/src/org/eclipse/emf/ecore/plugin/EcorePlugin.java

```diff
@@ -1309,4 +1337,361 @@ private static void computeModels(Map<URI, URI> pluginMap, Map<String, URI> nsUR
    * Since 2.14
    */
   public static final String ANNOTATION_VALIDATOR_PPID = "annotation_validator";
+
+  /**
+   * In case we are not running in Eclipse, this BundleActivator will be used.
+   */
+  private static class PlainOSGiBundleActivator implements BundleActivator
+  {
+    private ServiceTracker<EPackage.Registry, Object> osgiEPackageRegistryTracker;
+
+    @Override
+    public void start(BundleContext context) throws Exception
+    {
+      if (defaultRegistryImplementation == null)
+      {
+        OSGiDelegatEPackageRegistry osgiDelegateEPackageRegistry = new OSGiDelegatEPackageRegistry(context);
+        osgiEPackageRegistryTracker = new ServiceTracker<>(context, OSGiDelegatEPackageRegistry.FILTER, osgiDelegateEPackageRegistry);
+        osgiEPackageRegistryTracker.open();
+        defaultRegistryImplementation = osgiDelegateEPackageRegistry;
+      }
```

Comment:
This is effectively always be null.  So the specialized registry is effectively **always** created and the created and tracker is started.  I.e., there is no system property to guard this.